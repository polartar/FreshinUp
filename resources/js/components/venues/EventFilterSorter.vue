<template>
  <search-filter-sorter
    ref="filter"
    without-filter-label
    placeholder="Search"
    v-bind="$attrs"
    v-on="$listeners"
    @run="run"
  />
</template>

<script>
import SearchFilterSorter from 'fresh-bus/components/search/filter-sorter'
export default {
  components: {
    SearchFilterSorter
  },
  methods: {
    run (params) {
      this.$emit('runFilter', {
        sort: params.orderBy,
        'filter[name]': params.term
      })
    }
  }
}
</script>
