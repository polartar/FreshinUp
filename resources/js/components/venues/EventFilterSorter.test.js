import { mount, shallowMount } from '@vue/test-utils'
import Component from './EventFilterSorter.vue'
import * as Stories from './EventFilterSorter.stories'

describe('components/venues/EventFilterSorter', () => {
  describe('Snapshots', () => {
    test('Default', () => {
      const wrapper = mount(Stories.Default())
      expect(wrapper.element).toMatchSnapshot()
    })
    test('Set', () => {
      const wrapper = mount(Stories.Set())
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    describe('run(params)', () => {
      test('when empty', async () => {
        const wrapper = shallowMount(Component)
        const params = {}
        wrapper.vm.run(params)
        await wrapper.vm.$nextTick()
        const emitted = wrapper.emitted().runFilter
        expect(emitted).toBeTruthy()
        expect(emitted[0][0]).toMatchObject(params)
      })
      test('when set', async () => {
        const wrapper = shallowMount(Component)
        const params = {
          term: 'abc',
          orderBy: 'created_at'
        }
        wrapper.vm.run(params)
        await wrapper.vm.$nextTick()
        const emitted = wrapper.emitted().runFilter
        expect(emitted).toBeTruthy()
        expect(emitted[0][0]).toMatchObject({
          'filter[name]': params.term,
          sort: params.orderBy
        })
      })
    })
  })
})
