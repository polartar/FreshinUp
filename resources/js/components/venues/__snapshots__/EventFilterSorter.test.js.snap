// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/venues/EventFilterSorter Snapshots Default 1`] = `
<div
  class="container"
>
  <div
    class="v-card v-sheet theme--light"
    style="background-color: rgb(248, 248, 248); border-color: #f8f8f8;"
  >
    <form
      class="v-form"
      novalidate="novalidate"
    >
      <div
        class="container filter-sorter-container fluid"
      >
         
        <div
          class="layout filter-sorter-main-layout row wrap"
        >
          <!---->
           
          <div
            class="flex grow"
          >
            <div
              class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      placeholder="Search"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="flex shrink"
          >
            <div
              class="v-input filter-sorter-sortable-select v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-select theme--light"
              role="combobox"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-select__slot"
                  >
                    <label
                      aria-hidden="true"
                      class="v-label theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      Sort by
                    </label>
                    <div
                      class="v-select__selections"
                    >
                      <input
                        aria-label="Sort by"
                        aria-readonly="false"
                        autocomplete="on"
                        data-vv-name="sort"
                        readonly="readonly"
                        type="text"
                      />
                    </div>
                    <div
                      class="v-input__append-inner"
                    >
                      <div
                        class="v-input__icon v-input__icon--append"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          arrow_drop_down
                        </i>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-menu"
                  >
                    <div
                      class="v-menu__content theme--light "
                      style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                    >
                      <div
                        class="v-select-list v-card theme--light"
                      >
                        <div
                          class="v-list theme--light"
                          role="list"
                        >
                          <div
                            role="listitem"
                          >
                            <div
                              class="v-list__tile theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  No data available
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <!---->
           
        </div>
         
        <!---->
         
        <div
          class="layout filter-sorter-expanded-layout align-center justify-center row wrap fill-height"
          style="display: none;"
        >
          <div
            class="flex"
          />
           
          <div
            class="flex text-no-wrap"
          >
            <button
              class="v-btn v-btn--large theme--light"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
            Clear Filters
          
              </div>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`components/venues/EventFilterSorter Snapshots Set 1`] = `
<div
  class="container"
>
  <div
    class="v-card v-sheet theme--light"
    sort-options="[object Object],[object Object],[object Object],[object Object]"
    style="background-color: rgb(248, 248, 248); border-color: #f8f8f8;"
  >
    <form
      class="v-form"
      novalidate="novalidate"
    >
      <div
        class="container filter-sorter-container fluid"
      >
         
        <div
          class="layout filter-sorter-main-layout row wrap"
        >
          <!---->
           
          <div
            class="flex grow"
          >
            <div
              class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder theme--light"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <input
                      placeholder="Search"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="flex shrink"
          >
            <div
              class="v-input filter-sorter-sortable-select v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-select v-input--is-label-active v-input--is-dirty theme--light"
              role="combobox"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-select__slot"
                  >
                    <div
                      class="v-select__selections"
                    >
                      <div
                        class="v-select__selection v-select__selection--comma"
                      >
                        Oldest
                      </div>
                      <input
                        aria-label="Sort by"
                        aria-readonly="false"
                        autocomplete="on"
                        data-vv-name="sort"
                        readonly="readonly"
                        type="text"
                      />
                    </div>
                    <div
                      class="v-input__append-inner"
                    >
                      <div
                        class="v-input__icon v-input__icon--append"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon material-icons theme--light"
                        >
                          arrow_drop_down
                        </i>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-menu"
                  >
                    <div
                      class="v-menu__content theme--light "
                      style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                    >
                      <div
                        class="v-select-list v-card theme--light"
                      >
                        <div
                          class="v-list theme--light"
                          role="list"
                        >
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  Newest
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            class="primary--text"
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  Oldest
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  Name (A - Z)
                                </div>
                              </div>
                            </a>
                          </div>
                          <div
                            role="listitem"
                          >
                            <a
                              class="v-list__tile v-list__tile--link theme--light"
                            >
                              <div
                                class="v-list__tile__content"
                              >
                                <div
                                  class="v-list__tile__title"
                                >
                                  Name (Z - A)
                                </div>
                              </div>
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <!---->
           
        </div>
         
        <!---->
         
        <div
          class="layout filter-sorter-expanded-layout align-center justify-center row wrap fill-height"
          style="display: none;"
        >
          <div
            class="flex"
          />
           
          <div
            class="flex text-no-wrap"
          >
            <button
              class="v-btn v-btn--large theme--light"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
            Clear Filters
          
              </div>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
